### PR TITLE
op-guide: Add document for deploying TiDB with docker

### PR DESCRIPTION
**Pull request by shenli**
@siddontang @iamxy @qiuyesuifeng PTAL

### CODE REVIEW

File: op-guide/docker.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# 使用 Docker 构建和运行 TiDB
+
+本文档用于说明如何使用 Docker 部署 TiDB 服务
+
+## Docker 安装和环境准备
+
+#### 安装 Docker
+使用 Docker 可以非常快速搭建起一套 TiDB 环境。Docker 可以方便地在 Linux / Mac OS X / Windows 环境上安装，安装过程可参考 Docker 官网 https://docs.docker.com/linux/
```
Comment (qiuyesuifeng) [inline op-guide/docker.md]: Should we refer url only to `https://docs.docker.com`?

File: op-guide/docker.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+#### 准备环境
+完整的 TiDB 运行需要三个组件，TiDB, TiKV 和 PD。
+* tidb-server 是 TiDB 的执行进程，负责客户端的接入，对用户的 SQL 进行解析、优化和执行，分解成下层的 KV 操作，并将执行结果进行聚合返回给客户端。TiDB 对外兼容 MySQL 协议，可以直接使用 MySQL Client 进行测试
+* tikv-server，作为 TiDB 的分布式 KV 存储引擎，可以在网络互通的多机环境部署，使用 Raft 协议实现强一致性的数据复制，因此推荐 tikv-server 的部署节点为奇数个，当多数派节点存活 TiKV 保持可用状态
```
Comment (qiuyesuifeng) [inline op-guide/docker.md]: This description is not exact,  we can guarantee that if the store count is more than `max-peer-count` config,  the TiKV is  high available.

File: op-guide/docker.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+* pd-server，负责 TiKV 的 region 路由信息的维护，依托于 etcd 作为元数据的存储。并协调处理 region 的 rebalance 以及 merge 和 split 等操作
```
Comment (qiuyesuifeng) [inline op-guide/docker.md]: Now we have no etcd component, so should we remove etcd?

File: op-guide/docker.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+#### 启动 pd-server
+```
+docker run -d --name pd \
+  -p 1234:1234 \
+  -p 9090:9090 \
+  -p 2379:2379 \
+  -p 2380:2380 \
+  -v /usr/share/ca-certificates/:/etc/ssl/certs \
+  -v /etc/localtime:/etc/localtime:ro \
```
Comment (qiuyesuifeng) [inline op-guide/docker.md]: What is this volume used for?

File: op-guide/docker.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+docker run -d --name tidb-server -v /etc/localtime:/etc/localtime:ro -p 4000:4000 -p 10080:10080 pingcap/tidb:latest -L info --path /tmp/tidb
+```
+
+这样就在 4000 端口上启动了 tidb-server，可以使用 mysql client 连接 tidb-server:
+```
+mysql -h 127.0.0.1 -P 4000 -u root -D test
+```
+
+## 使用 TiKV 存储引擎
+使用 TiKV 作为存储引擎可以构建分布式数据库。
```
Comment (qiuyesuifeng) [inline op-guide/docker.md]: Alignment?

File: op-guide/docker.md (hunk 4 above)
Comment (shenli) [inline op-guide/docker.md]: I am not sure. @iamxy

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by shenli** [comment]
@siddontang PTAL

File: op-guide/docker.md (hunk 4 above)
Comment (iamxy) [inline op-guide/docker.md]: -v /etc/localtime:/etc/localtime:ro, this set timezone in docker as same as in host.

File: op-guide/docker.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+使用 docker 可以直接启动：
+```
+docker run -d --name tidb-server \
+  -v /etc/localtime:/etc/localtime:ro \
+  -p 4000:4000 \
+  -p 10080:10080 \
+  pingcap/tidb:latest \
+  -L info --path /tmp/tidb
+
+docker run -d --name tidb-server -v /etc/localtime:/etc/localtime:ro -p 4000:4000 -p 10080:10080 pingcap/tidb:latest -L info --path /tmp/tidb
```
Comment (queenypingcap) [inline op-guide/docker.md]: 这个命令跟上面的命令是重复的。It's my fault. 我当时看这个文档的时候自己写上的，我马上去源文档把这个删了

Comment (shenli) [inline op-guide/docker.md]: OK, I will delete it.

**Comment by shenli** [comment]
@qiuyesuifeng @iamxy @siddontang PTAL

File: op-guide/docker.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+## Docker 安装和环境准备
+
+#### 安装 Docker
+使用 Docker 可以非常快速搭建起一套 TiDB 环境。Docker 可以方便地在 Linux / Mac OS X / Windows 环境上安装，安装过程可参考 Docker 官网 https://docs.docker.com/ 。
+
+#### 准备环境
+完整的 TiDB 运行需要三个组件，TiDB, TiKV 和 PD。
+* tidb-server 是 TiDB 的执行进程，负责客户端的接入，对用户的 SQL 进行解析、优化和执行，分解成下层的 KV 操作，并将执行结果进行聚合返回给客户端。TiDB 对外兼容 MySQL 协议，可以直接使用 MySQL Client 进行测试
+* tikv-server，作为 TiDB 的分布式 KV 存储引擎，可以在网络互通的多机环境部署，使用 Raft 协议实现强一致性的数据复制，因此推荐设置 Region 的最大复本数（max-peer-count）为奇数，当多数派节点存活时 TiKV 保持可用状态
```
Comment (siddontang) [inline op-guide/docker.md]: Now we have removed max-peer-count in flag
need re-add? @qiuyesuifeng

File: op-guide/docker.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
+# 使用 Docker 构建和运行 TiDB
+
+本文档用于说明如何使用 Docker 部署 TiDB 服务
```
Comment (qiuyesuifeng) [inline op-guide/docker.md]: end of 。

File: op-guide/docker.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+## Docker 安装和环境准备
+
+#### 安装 Docker
+使用 Docker 可以非常快速搭建起一套 TiDB 环境。Docker 可以方便地在 Linux / Mac OS X / Windows 环境上安装，安装过程可参考 Docker 官网 https://docs.docker.com/ 。
+
+#### 准备环境
+完整的 TiDB 运行需要三个组件，TiDB, TiKV 和 PD。
+* tidb-server 是 TiDB 的执行进程，负责客户端的接入，对用户的 SQL 进行解析、优化和执行，分解成下层的 KV 操作，并将执行结果进行聚合返回给客户端。TiDB 对外兼容 MySQL 协议，可以直接使用 MySQL Client 进行测试
```
Comment (qiuyesuifeng) [inline op-guide/docker.md]: end of 。

File: op-guide/docker.md (hunk 7 above)
Comment (qiuyesuifeng) [inline op-guide/docker.md]: end of 。

**Comment by shenli** [comment]
PTAL @qiuyesuifeng @siddontang

File: op-guide/docker.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+## Docker 安装和环境准备
+
+#### 安装 Docker
+使用 Docker 可以非常快速搭建起一套 TiDB 环境。Docker 可以方便地在 Linux / Mac OS X / Windows 环境上安装，安装过程可参考 Docker 官网 https://docs.docker.com/ 。
+
+#### 准备环境
+完整的 TiDB 运行需要三个组件，TiDB, TiKV 和 PD。
+* tidb-server 是 TiDB 的执行进程，负责客户端的接入，对用户的 SQL 进行解析、优化和执行，分解成下层的 KV 操作，并将执行结果进行聚合返回给客户端。TiDB 对外兼容 MySQL 协议，可以直接使用 MySQL Client 进行测试。
+* tikv-server，作为 TiDB 的分布式 KV 存储引擎，可以在网络互通的多机环境部署，使用 Raft 协议实现强一致性的数据复制，因此推荐设置 Region 的最大复本数（max-peer-count）为奇数，当多数派节点存活时 TiKV 保持可用状态。
```
Comment (siddontang) [inline op-guide/docker.md]: same here, we don't have max-peer-count here now. 
@qiuyesuifeng

Comment (shenli) [inline op-guide/docker.md]: PATL @siddontang

**Comment by siddontang** [comment]
LGTM

File: op-guide/docker.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+  --cluster-id=1
+```
+
+#### 启动 tikv-server
+在三台机器上分别启动 TiKV 服务
+```
+# 机器1
+docker run -d --name tikv1 \
+  -p 20160:20160 \
+  -v /etc/localtime:/etc/localtime:ro \
```
Comment (qiuyesuifeng) [inline op-guide/docker.md]: Alignment

Comment (shenli) [inline op-guide/docker.md]: @qiuyesuifeng PTAL

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**